### PR TITLE
Documentation Extension

### PR DESCRIPTION
**Pull request by eprei**
* Addition of a project overview.

* Breakdown of the various windows, functionalities and graphical elements necessary for the realization of the project.

* In addition to reading the diff files from github, the modifications can be seen in a nicer way in the docs/index.html file or here https://eprei.github.io/ft_transcendence/

### CODE REVIEW

File: design/design.adoc
Comment (t-h2o) [inline design/design.adoc:1]: is it an extra file ?
if yes, delete it with `git rm file`

File: docs/README.adoc [hunk 1]
```diff
@@ -35,3 +36,198 @@ portin_react -up-> portout_react
...
+	object Welcome {
+		Login with the intranet OAuth system 42
+		Communication with OAuth 42
+	}
+
+	object Profile {
+		Choose username
+		Choose/upload an avatar
+		Enable 2FA
+		Add users ass friends and view real time status
```
Comment (t-h2o) [inline docs/README.adoc:186]: typo ?
```suggestion
		Add users as friends and view real time status
```
Split this sentence:
```suggestion
		Add users as friends
		See the status of friends in real time
```

File: docs/README.adoc [hunk 2]
```diff
@@ -35,3 +36,198 @@ portin_react -up-> portout_react
...
+image::assets/Page1.1_Welcome.png[welcome preview]
+
+=== Page 2: Compte utilisateur
+
+==== 2.1 Profile
+
+==== Functionalities:
+
+* You can choose a (unique) username to be displayed on the site. We start with the default name taken from OAuth 42 (?)
+* You can upload an avatar. If you don't, it defaults to your 42 profile picture (?)
```
Comment (t-h2o) [inline docs/README.adoc:65]: better ?
```suggestion
* You can upload an avatar. If you don't, it is by default to your 42 profile picture (?)
```

File: docs/assets/Page3_Chat.png
Comment (t-h2o) [inline docs/assets/Page3_Chat.png:1]: By consistency, can you rename this file by `Page3.1_Chat.png` ?

File: docs/README.adoc (hunk 1 above)
Comment (eprei) [reply docs/README.adoc:186]: Done

File: docs/README.adoc [hunk 3]
```diff
@@ -35,3 +36,198 @@ portin_react -up-> portout_react
...
+=== Window scheme
+
+[plantuml, target=assets/design, format=svg, width=100%]
+....
+rectangle {
+
+	object Page01
+	object Page02
+	object Page03
+	object Page04
```
[the remark this answers is not in the document]
Comment (eprei) [reply docs/README.adoc:175]: Nice method

[the remark this answers is not in the document]
Comment (t-h2o) [reply docs/README.adoc]: > What is the most efficient way to proceed now?

IMHO, the most efficient is to do a commit per comment
and to reply to the comment with the commit ID,
this way, the commenter can check if the commit matches the expectations.

File: docs/README.adoc [hunk 4]
```diff
@@ -35,3 +36,199 @@ portin_react -up-> portout_react
...
+		CosmicPong (logo)
+		Profil
+		Match History
+		Chat
+		The game
+		LogOut (?)
+		42Lausanne (logo)
+	}
+
+	object TheGame{
```
Comment (t-h2o) [inline docs/README.adoc:213]: - Adding space ?
- Add a space before the curly brace `{` ?
```suggestion
	object "The game" as TheGame {
```

File: docs/README.adoc [hunk 5]
```diff
@@ -1,6 +1,7 @@
 = Transcendence
 :nofooter:
```
Comment (t-h2o) [inline docs/README.adoc:2]: Resource: [active the table of contents](https://docs.asciidoctor.org/asciidoc/latest/toc/#activate-the-toc)
```suggestion
:nofooter:
:toc: left
```

**Review by fire-poy — APPROVED** [review]
Great documentation! awesome job!

File: docs/README.adoc (hunk 5 above)
Comment (shaolin-peanut) [reply docs/README.adoc:2]: that's a good idea

**Review by shaolin-peanut — APPROVED** [review]
LGTM

**Review by RobinBurri — APPROVED** [review]
Nice design.  Top

File: docs/README.adoc (hunk 5 above)
Comment (eprei) [reply docs/README.adoc:2]: Stylé

**Review by t-h2o — APPROVED** [review]

**Pull request merged**